### ISSUE TITLE
JSON encoding API usage with usage of safe factory

### ISSUE THREAD

**Issue by rbairwell** [kalessil/phpinspectionsea#1428]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         |  Php Inspections (EA Ultimate) 3.0.4.2 |
| Language level | PHP 7.4beta4                        |
| Inspection | JSON encoding API usage |

Current behaviour
--
With the code:
```php
<?php
declare(strict_types=1);
include 'vendor/autoload.php';
use function Safe\json_encode;

print json_encode('test');
```
I would not expect the `Please consider taking advantage of JSON_THROW_ON_ERROR flag for this operation` message to appear as `json_encode` has been overriden by https://github.com/thecodingmachine/safe/blob/master/generated/json.php#L45 which will handle this.

Expected behaviour
--
The inspection is not triggered if `json_encode` is overriden (and ideally detects if the overriding function does a check).

Environment details
---
PhpStorm 2019.2.3
Build #PS-192.6817.20, built on September 25, 2019
Runtime version: 11.0.4+10-b304.69 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Windows 10 10.0
GC: ParNew, ConcurrentMarkSweep
Memory: 972M
Cores: 8
Registry: run.processes.with.pty=TRUE
Non-Bundled Plugins: GrepConsole, Key Promoter X, PHPUnit code coverage, String Manipulation, com.intellij.ideolog, net.seesharpsoft.intellij.plugins.csv, NEON support, com.kalessil.phpStorm.phpInspectionsUltimate, de.espend.idea.php.phpunit, de.espend.idea.php.toolbox, de.espend.idea.php.annotation, izhangzhihao.rainbow.brackets, com.intellij.plugin.adernov.powershell, org.psliwa.idea.composer, ru.adelf.idea.dotenv

**Comment by kalessil** [comment]
Fixed!
